internal/preflight/dryrun.go

```diff
@@ -19,9 +19,15 @@ import (
 
 type DryRun struct {
 	client client.Writer
+	log    logr.Logger
 }
 
-func NewDryRun(client client.Writer) *DryRun { return &DryRun{client: client} }
+func NewDryRun(client client.Writer, log logr.Logger) *DryRun {
```

Comment:
I see that you added a Logger as parameter to several functions that had no Context. While I prefer that instead of just hiding it in the Context I would like to discuss how we can make this consistent. I would argue that it is better to either:
- Only get the Logger from Context. If a method needs a logger but it gets no Context it needs to be rewritten to receive a context.
- Always pass the logger explicitly, never getting it from the Context.

What do you think? (also @erdii )

Reply:
If we use option 1 here, what is the exact update we can have? Is this the correct way to update it?
```
log, err := logr.FromContext(ctx)
	if err != nil {
		return fmt.Errorf("no logger in context")
	}
```
For the controller runtime, we can use `ctrl.LoggerFrom(ctx)`, but here if we want to solve the missing logger to inject the log is a better way?